### DISCUSSION TITLE
Terminal becomes transparent when setting the background image

### DISCUSSION THREAD

**Discussion by VPraharsha03** [wezterm/wezterm#2906] [Q&A] [answered]
I'm trying to mimic the [Aurelia Retro Theme](https://github.com/mobilemancer/windows-terminal-aurelia), while doing so when I'm aligning the transparent png to the bottom right, the entire terminal becomes transparent. To prevent that I tried adding a solid color/gradient layer before the png layer, but there was no effect, the terminal still appeared to be transparent. 

Here's my config:
```lua
local wezterm = require 'wezterm'
return 
{
	background = 
	{
		{
			source =
			{
				--Color = 'red',
				Gradient = 
				{
					orientation = 'Vertical',
					colors = 
					{
						'#0f0c29',
						'#302b63',
						'#24243e',
					},
					interpolation = 'Linear',
					blend = 'Rgb',
				},
			},
		},
		{
			source = 
			{
				File = 'au-os-M.png',
			},
			vertical_align = 'Bottom',
			horizontal_align = 'Right',
			height = '50%',
			width = '35%',
		},
	},
}
```
Is there any other way to do so?

**Comment by wez** [comment, marked as answer]
When I ran that config, I see this error message about the gradient:

```
08:17:18.786  ERROR  wezterm_gui::termwindow::background > Failed to load background: Cover is not implemented for background gradients. Use e.g. `width = '100%'` instead
```

(you can also review error logs by pressing `CTRL-SHIFT-L` to open the debug overlay).

Add:

```lua
      height = '100%',
      width = '100%',
```

to your gradient to have it fill the window.

As for the background becoming transparent when you set a PNG into it: that behavior was intentional for the original singular `window_background_image` option. It may need some additional consideration when used in the newer `background` setting.

**Comment by VPraharsha03** [comment]
> (you can also review error logs by pressing CTRL-SHIFT-L to open the debug overlay).

Cool, the debug overlay is a really handy feature. Thank you, that fixes both of the problems.

I'd one more question, how do I use Windows Environment variables ? using them directly in the config seems to be wrong:
```
ERROR wezterm_gui::termwindow::background > Failed to load background: getting metadata for %USERPROFILE%\.config\wezterm\au-os-M.png: The system cannot find the path specified. (os error 3)
```

**Reply by wez** [reply]
The built-in lua function [os.getenv()](https://www.lua.org/manual/5.3/manual.html#pdf-os.getenv) can be used to read environment variables.

As a special wezterm extension over the basic lua `os.getenv` function, on Windows, wezterm will also pick up environment variables from the registry without needing to be restarted.
